src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx

```diff
@@ -198,24 +201,6 @@ describe('WorkspaceDetail', () => {
     expect(document.querySelector('#dataSources')).toHaveClass('euiTab-isSelected');
   });
 
-  it('click on delete button will show delete modal', async () => {
-    const workspaceService = createWorkspacesSetupContractMockWithValue(workspaceObject);
-    const { getByText, getByTestId, queryByText } = render(
-      WorkspaceDetailPage({ workspacesService: workspaceService })
-    );
-    fireEvent.click(getByText('delete'));
-    expect(getByText('Delete workspace')).toBeInTheDocument();
-    fireEvent.click(getByText('Cancel'));
-    expect(queryByText('Delete workspace')).toBeNull();
-    fireEvent.click(getByText('delete'));
-    const input = getByTestId('delete-workspace-modal-input');
-    fireEvent.change(input, {
-      target: { value: 'delete' },
-    });
-    const confirmButton = getByTestId('delete-workspace-modal-confirm');
-    fireEvent.click(confirmButton);
-  });
```

Comment:
Is there any way that we can keep this unit test?

Reply:
Due to the inability to render this delete button, we can only judge whether this test id exists on the page.

Reply:
Actually the unit test is covering the delete flow for our features. I just feel a little bit uncomfortable that we delete it.

The worst case is that we can mock the HeaderControl component to execute the `run` function to popup the modal.

Reply:
Sure, in order to cover the delete flow, I execute the `run` function to show the delete modal by `controls[0].run();`